Example/ResizableView.swift

```diff
@@ -0,0 +1,141 @@
+        imageView.isUserInteractionEnabled = true
+        imageView.layer.shadowColor = #colorLiteral(red: 0.1029271765, green: 0.08949588804, blue: 0.1094761982, alpha: 0.8005611796)
+        imageView.layer.shadowOffset = CGSize(width: 0, height: 1)
+        imageView.layer.shadowOpacity = 1
+        imageView.layer.shadowRadius = 1.0
+        
+        panRecognizer.require(toFail: resizePanRecognizer)
+        
+        addGestureRecognizer(panRecognizer)
+        imageView.addGestureRecognizer(resizePanRecognizer)
+        addSubview(imageView)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    @objc func pan(_ sender: UIPanGestureRecognizer) {
+        if sender.state == .began || sender.state == .changed {
+            center = sender.location(in: superview)
+            
+            layoutSubviews()
+        }
+    }
+    
+    @objc func resizePan(_ sender: UIPanGestureRecognizer) {
+        
+        let location = sender.location(in: superview)
+        
+        if sender.state == .began || sender.state == .changed {
+            
+            let origin = CGPoint(x: frame.minX, y: frame.minY)
+            frame = CGRect(origin: origin,
+                           size: CGSize(width: location.x.distance(origin.x),
+                                        height: location.y.distance(origin.y)))
+            layoutSubviews()
+        }
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        if lineLayer.superlayer == nil {
+            lineLayer.strokeColor = #colorLiteral(red: 0, green: 0.5490196078, blue: 1, alpha: 1).cgColor
+            lineLayer.fillColor = UIColor.clear.cgColor
+            lineLayer.lineWidth = 1
+            lineLayer.lineDashPattern = [5, 5]
+            layer.addSublayer(lineLayer)
+        }
+        
+        lineLayer.path = UIBezierPath(rect: bounds).cgPath
+        
+        let clippedPath = UIBezierPath(rect: superview!.bounds)
+        clippedPath.append(UIBezierPath(rect: lineLayer.frame))
+        
+        let superFrame = self.convert(superview!.bounds, to: self)
+        
+        if let backgroundLayer = backgroundLayer {
+            backgroundLayer.path = UIBezierPath(rect: superFrame).cgPath
+            backgroundLayer.frame = superFrame
+            
+            let path = UIBezierPath(rect: frame)
+            path.append(UIBezierPath(rect: backgroundLayer.bounds))
+            
+            maskLayer.fillRule = kCAFillRuleEvenOdd
+            maskLayer.path = path.cgPath
+            backgroundLayer.mask = maskLayer
+        }
+        
+        imageView.center = CGPoint(x: bounds.maxX-5, y: bounds.maxY-5)
+        
+        bringSubview(toFront: imageView)
+    }
+    
+}
+
+fileprivate extension CGFloat {
+    
+    var mid: CGFloat {
+        return self / 2
+    }
+    
+    func distance(_ to: CGFloat) -> CGFloat {
```

Comment:
The calculations in the pan gesture recognizer were getting unreadable but it has been cleaned up since then so it makes sense to remove this simple distance.